lib/model/folder_sendrecv.go

```diff
@@ -848,6 +848,11 @@ func (f *sendReceiveFolder) deleteFileWithCurrent(file, cur protocol.FileInfo, h
 		dbUpdateChan <- dbUpdateJob{file, dbUpdateDeleteFile}
 		return
 	}
+
+	if err = osutil.TraversesSymlink(f.fs, filepath.Dir(file.Name)); err != nil {
```

Comment:
Shouldn't we maybe record the deletion to the database here, like in the `!hasCur` case above? If the files is behind a symlink it should never have existed to begin with in our viewpoint and we may as well consider it deleted?

Reply:
Right. I updated the test to expect a db update, but check that the actual file behind the symlink has been left alone.